hw1-app/src/app/core/auth.service.ts

```diff
@@ -0,0 +1,37 @@
+import {Injectable} from '@angular/core';
+import {Router} from '@angular/router';
+import {Subject} from 'rxjs';
+
+@Injectable()
+export class AuthService {
+  public isAuthenticated = false;
+  public authSubject = new Subject();
```

Comment:
better way to do subject private, to remove access for him outside

Reply:
private authSubject = new Subject<any>();
 
  // Observable streams
  authObs$ = this.authSubject.asObservable();